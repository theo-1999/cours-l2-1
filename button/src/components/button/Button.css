.toggled {
  border: 1px solid red;
  background-color: white;
}
.untoggled {
  border: 1px solid blue;
  background-color: violet;
}

/*.bouton {
  background-color: violet;
  height: 400px;
  width: 400px;
  
}
*/