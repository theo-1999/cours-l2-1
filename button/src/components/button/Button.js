import React from "react";
import './Button.css';

class Button extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
       toggled : false
       
    }
    this.handleClick = this.handleClick.bind(this);
      // ...
  }

  handleClick() {
   this.setState( (oldState) => {
     const newState = {
       toggled : !oldState.toggled
     }
     //..
     console.log(newState);
     return newState; 
     
   });
  }

  render() {
    return (
      <button className={ this.state.toggled ? "toggled" : "untoggled" }
         data-testid="button"
         
      onClick={this.handleClick}>
      {this.props.children ? this.props.children : "Add text!"}
      </button>);
  }

}


export default Button;
