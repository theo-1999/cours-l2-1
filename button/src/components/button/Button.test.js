import React from 'react';
import ReactDOM from 'react-dom';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import Button from './Button';

let container = null;

beforeEach( () => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach( () => {
  unmountComponentAtNode(container);
  container.remove();
});

it('renders without crashing', () => {
  act( () => {
    render(<Button>Hello, world!</Button>, container);
  });
  expect(document.querySelector("[data-testid='button']").textContent).toMatch(
    "Hello, world!"
  );

  act( () => {
    render(<Button></Button>, container);
  });
  expect(document.querySelector("[data-testid='button']").textContent).toMatch(
    "Add text"
  );
});

it("changes CSS class when we click on it", () => {
  act( () => {
    render(<Button></Button>, container);
  });
  const button = document.querySelector("[data-testid='button']");
  expect(button.className).toBe("untoggled");
  
  act( () => {
    button.dispatchEvent(
      new MouseEvent("click", { bubbles: true })
    );
  });
  expect(button.className).toBe("toggled");
});

it("corresponds to what we saw before", () => {
  act( () => {
    render(<Button></Button>, container);
  });
  const button = document.querySelector("[data-testid='button']");
  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
