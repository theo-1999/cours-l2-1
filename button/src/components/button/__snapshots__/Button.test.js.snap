// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`corresponds to what we saw before 1`] = `"<button class=\\"untoggled\\" data-testid=\\"button\\">Add text!</button>"`;
